src/microsoft-extensions-ai/abstraction-implementations/AbstractionImplementationExamples/DependencyInjection.cs

```diff
@@ -0,0 +1,34 @@
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+public partial class AbstractionSamples
+{
+    public static async Task DependencyInjection() 
+    {
+        var app = Host.CreateApplicationBuilder();
+
+        app.Services.AddSingleton<IDistributedCache, InMemoryCacheStorage>();
+        app.Services.AddChatClient(builder => {
+            var endpoint = new Uri("http://coolsite.ai");
+            var modelId = "my-custom-model";
+
+            var cache = builder.Services.GetRequiredService<IDistributedCache>();
```

Comment:
This is not needed. You can delete this line and make the `.UseDistributedCache()` call below be parameterless.